File: go/1100/D1742.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func gcd(a, b int) int {
	for b != 0 {
		a, b = b, a%b
	}
	return a
}

func main() {
	in := bufio.NewReaderSize(os.Stdin, 1<<20)
	out := bufio.NewWriterSize(os.Stdout, 1<<20)
	defer out.Flush()

	var t int
	fmt.Fscan(in, &t)

	for t > 0 {
		t--
		var n int
		fmt.Fscan(in, &n)
		pos := make([]int, 1001)
		for i := range pos {
			pos[i] = 0
		}
		for i := 1; i <= n; i++ {
			var x int
			fmt.Fscan(in, &x)
			pos[x] = i
		}
		ans := -1
		for x := 1; x <= 1000; x++ {
			for y := 1; y <= 1000; y++ {
				if pos[x] > 0 && pos[y] > 0 && gcd(x, y) == 1 {
					ans = max(ans, pos[x]+pos[y])
				}
			}
		}
		fmt.Fprintln(out, ans)
	}
}

File: 1100/random/D1742.cpp
#include <iostream>
#include <vector>
#include <numeric>
using namespace std;

int main() {
    ios::sync_with_stdio(0); cin.tie(0);
    int T; cin>>T;
    while (T--) {
        int n; cin>>n;
        vector<int> pos(1001);
        for (int i=0;i<n;i++) {
            int x; cin>>x;
            pos[x]=i+1;
        }
        int ans=-1;
        for (int i=1;i<=1000;i++) {
            for (int j=1;j<=1000;j++) {
                if (pos[i] && pos[j] && gcd(i,j)==1) {
                    ans=max(ans,pos[i]+pos[j]);
                }
            }
        }
        cout<<ans<<'\n';
    }
    return 0;
}